src/map_to_curve.rs

```diff
@@ -56,6 +56,47 @@ impl GroupWithMapToCurve for RistrettoPoint {
     }
 }
 
+#[cfg(feature = "p256")]
+impl GroupWithMapToCurve for p256_::ProjectivePoint {
+    const SUITE_ID: usize = 0x0003;
+
+    fn map_to_curve<H: Hash>(msg: &[u8], dst: &[u8]) -> Result<Self, ProtocolError> {
+        // https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-11#section-8.2
+        // `P256_XMD:SHA-256_SSWU_RO_` has an `L` of `48`
+        const L: usize = 48;
+
+        // https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-11#section-3
+        // `hash_to_curve` calls `hash_to_field` with a `count` of `2`
+        // https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-11#section-5.3
+        // `hash_to_field` calls `expand_message` with a `len_in_bytes` of `count * L`
+        let uniform_bytes = expand_message_xmd::<H>(msg, dst, 2 * L)?;
+
+        <Self as Group>::hash_to_curve(&GenericArray::clone_from_slice(&uniform_bytes[..]))
+            .map_err(ProtocolError::from)
+    }
+
+    fn hash_to_scalar<H: Hash>(input: &[u8], dst: &[u8]) -> Result<Self::Scalar, ProtocolError> {
+        use num_bigint::BigUint;
+        use num_integer::Integer;
+        use std::str::FromStr;
+
+        // https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-11#section-8.2
+        // `p: 2^256 - 2^224 + 2^192 + 2^96 - 1`
+        let p = BigUint::from_str(
+            "115792089210356248762697446949407573530086143415290314195533631308867097853951",
+        )
+        .unwrap();
+        // `P256_XMD:SHA-256_SSWU_RO_` has an `L` of `48`
+        const L: usize = 48;
+        let uniform_bytes = expand_message_xmd::<H>(input, dst, L)?;
+        let bytes = BigUint::from_bytes_be(&uniform_bytes).mod_floor(&p);
+
+        Ok(p256_::Scalar::from_bytes_reduced(GenericArray::from_slice(
+            &bytes.to_bytes_be(),
+        )))
```

Comment:
This is the offending piece of code right now, I can't get this to work. According to all our tests `expand_message_xmd` works correctly, so I can only believe that `mod_floor(p)` is nonsense, or `from_bytes_reduced` just doesn't work here. Asking some questions in RustCrypto I was informed that there is a missing function `from_bytes_wide` that might be what I'm looking for. After many hours I finally gave up and decided implementing this function myself goes pretty much over my head, any help would be amazing.

Reply:
I tried to re-follow the steps I did when I tried to debug this yesterday and realized I forgot to mention, that I couldn't confirm that there isn't something wrong earlier, or to be more precise, inside `opaque::oprf_key_from_seed()`, which translates to `DeriveKeyPair` from the ORPF draft.

Reply:
I fixed it, I was using the wrong variable, I used `P`, but I should have used `n`.